packages/core-transactions/src/handlers/transaction.ts

```diff
@@ -105,24 +103,48 @@ export abstract class TransactionHandler implements ITransactionHandler {
         this.revertForRecipient(transaction, walletManager);
     }
 
-    protected applyToSender(transaction: Interfaces.ITransaction, walletManager: State.IWalletManager): void {
+    public applyToSender(transaction: Interfaces.ITransaction, walletManager: State.IWalletManager): void {
         const sender: State.IWallet = walletManager.findByPublicKey(transaction.data.senderPublicKey);
-        sender.balance = sender.balance.minus(transaction.data.amount).minus(transaction.data.fee);
+        const data: Interfaces.ITransactionData = transaction.data;
+
+        if (Utils.isException(data)) {
+            walletManager.logger.warn(
+                `Transaction forcibly applied as an exception: ${transaction.id}.`
+            );
+        }
+
+        this.canBeApplied(transaction, sender, walletManager);
+
+        if (data.version > 1) {
+            if (!sender.nonce.plus(1).isEqualTo(data.nonce)) {
+                throw new UnexpectedNonceError(data.nonce, sender.nonce, false);
+            }
+
+            sender.nonce = data.nonce;
+        }
+
+        const newBalance: Utils.BigNumber = sender.balance.minus(data.amount).minus(data.fee);
+        assert(!newBalance.isNegative());
```

Comment:
```suggestion
        assert(newBalance.isPositive());
```

Reply:
Positive is not the same as not negative.

Reply:
Marking this as resolved without addressing it. Assuming wrong suggestion.